runtimes/node-18.0/server.js

```diff
@@ -1,85 +1,125 @@
-const path = require("path");
+const fs = require("fs");
 const micro = require("micro");
-const { json, send } = require("micro");
+const { text, json, send } = require("micro");
 
 const USER_CODE_PATH = '/usr/code-start';
 
 const server = micro(async (req, res) => {
-    const body = await json(req);
-
-    if (req.headers[`x-internal-challenge`] !== process.env['INTERNAL_RUNTIME_KEY']) {
-        return send(res, 500, {stderr: 'Unauthorized'});
+    if (req.headers[`x-openruntimes-secret`] !== process.env['INTERNAL_RUNTIME_KEY']) {
+        return send(res, 500, 'Unauthorized');
     }
 
-    const request = {
-        variables: body.variables ?? {},
-        headers: body.headers ?? {},
-        payload: body.payload ?? '',
-    };
-
-    console.stdlog = console.log.bind(console);
-    console.stderror = console.error.bind(console);
-    console.stdinfo = console.info.bind(console);
-    console.stddebug = console.debug.bind(console);
-    console.stdwarn = console.warn.bind(console);
-
     const logs = [];
     const errors = [];
 
-    console.log = console.info = console.debug = console.warn = function() {
-        const args = [];
-        Array.from(arguments).forEach(arg => {
-            if(arg instanceof Object || Array.isArray(arg)) {    
-                args.push(JSON.stringify(arg));
-            } else {
-                args.push(arg)
-            }
-        });
-        logs.push(args.join(" "));
-    }
+    const contentType = req.headers['content-type'] ?? 'text/plain';
+    const body = contentType === 'application/json' ? await json(req) : await text(req);
 
-    console.error = function() {
-        const args = [];
-        Array.from(arguments).forEach(arg => {
-            if(arg instanceof Object || Array.isArray(arg)) {    
-                args.push(JSON.stringify(arg));
-            } else {
-                args.push(arg)
-            }
-        });
-        errors.push(args.join(" "));
-    }
+    const headers = {};
+    Object.keys(req.headers).filter((header) => !header.startsWith('x-openruntimes-')).forEach((header) => {
+        headers[header] = req.headers[header];
+    });
+
+    const context = {
+        req: {
+            body: body,
+            headers: headers,
+            method: req.method,
+            url: req.url
+        },
+        res: {
+            headers: {},
+            body: '',
+            status: 200,
+
+            send: function (body, status, headers) {
```

Comment:
I wonder if we want to add more access methods to make the interface even nicer, maybe take some inspiration from Utopia.

Reply:
Compared to utipia, we are missing: html(), jsonp() anf iframe()

From those, I would agree we should add `html`. Tho I would prefer to avoid others until community requests them. I would prefer to keep it simple. In the end, it's all just alias for `send()`.
WDYT? @eldadfux

Reply:
I was more in the direction of getHeader, addHeader and stuff like that